src/urllib3/response.py

```diff
@@ -934,6 +938,9 @@ def read(
                     )
                 return data
 
+            flush_decoder = not data
```

Comment:
potential functional change in combination with https://github.com/urllib3/urllib3/pull/3262/files/c49227eb983300c512098809105c1050fea1f25d#r1460291175

(possibly should become `(amt != 0 and not data)` )